### PR TITLE
script for grammar

### PR DESCRIPTION
**Pull request by s-cork**
I had a go at writing a script to generate the grammar

### CODE REVIEW

File: src/parser/generated_parser.ts [hunk 1]
```diff
@@ -2786,11 +2786,11 @@ export class GeneratedParser extends Parser {
         let a, b, invalid_kwarg, literal, mark;
         mark = this.mark();
         if ((a = this.name()) && (literal = this.expect("=")) && (b = this.expression())) {
-            return pegen.keyword_or_starred(this, new astnodes.keyword(a.id, b, ...EXTRA), true);
+            return pegen.keyword_or_starred(this, new astnodes.keyword(a.id, b, ...EXTRA), 1);
```
Comment (s-cork) [inline src/parser/generated_parser.ts:2789]: not quite there...

**Comment by albertjan** [comment]
it's less lines, but also harder to understand, I'm still on the fence on what the best approach is

File: tools/gen_parser/skulpt_parser_genarator.py [hunk 2]
```diff
@@ -73,54 +73,20 @@
...
-        # .replace("AugOperator *", "AugOperator")
-        .replace("RAISE", "pegen.RAISE")
-        .replace("NEW_TYPE_COMMENT", "pegen.NEW_TYPE_COMMENT")
-    )
-    # action = re.compile(r"^([A-Z]+[a-z]+[\w]*) ").sub(r" new astnodes.\1 ", action)  # astnode constructor
-    action = re.compile(r"\( ([bt]) \) \? \( \( expr_ty \) [bt] \)(.\w+) : null").sub(r"\1\2", action)
-    action = re.sub(r"pegen.augoperator \( (new astnodes.\w+) \)", r"\1", action)
-    action = re.sub(r"new astnodes.Py_([None|False|True|Ellipsis])", r"py\1", action)
+    # actions are separated by spaces and ?. is a single unit in javascript
+    action = action.replace("? .", "?.")
```
Comment (s-cork) [inline tools/gen_parser/skulpt_parser_genarator.py:89]: `b ? b.id : null` vs `b?.id` reference.

Needed this line to account for javascript not liking `? .` 
(All actions get parsed with spaces between tokens)

**Comment by s-cork** [comment]
closing this - it's here if we ever need it

**Pull request closed without merging**